pkg/flow/internal/controller/loader.go

```diff
@@ -164,6 +165,26 @@ func (l *Loader) populateGraph(g *dag.Graph, blocks []*ast.BlockStmt) diag.Diagn
 				continue
 			}
 
+			if registration.Singleton && block.Label != "" {
+				diags.Add(diag.Diagnostic{
+					Severity: diag.SeverityLevelError,
+					Message:  fmt.Sprintf("Singleton component %q must not have a label", componentName),
+					StartPos: block.NamePos.Position(),
+					EndPos:   block.NamePos.Add(len(componentName) - 1).Position(),
```

Comment:
Do we have the position of the label available? It might be useful to point to the label instead of the component name

Reply:
Hm, not directly, but we can offset for the component name, and component name + label (including the quotation marks), and it seems to be working nicely
```
StartPos: block.NamePos.Add(len(componentName) + 1).Position(),
EndPos:   block.NamePos.Add(len(componentName) + 1).Add(len(block.Label) + 1).Position(),

Error: ./example-config.river:16:38: Component "prometheus.integration.node_exporter" does not support labels

15 |
16 | prometheus.integration.node_exporter "foo" {
   |                                      ^^^^^
17 | }
```

Reply:
That won't always work depending on spacing, I don't think. Do you want to try updating the AST + parser to include the label position?

It's probably something we generally need, I remember a few times where I wish I had access to the label position but didn't.

Reply:
Yes! I see that you're already setting the LabelPos when you're parseBlockName, so it's a one-line change.

I'll see that it get properly handled in other places as well and update this PR!